validator-api/src/contract_cache/mod.rs

```diff
@@ -92,45 +98,96 @@ impl<C> ValidatorCacheRefresher<C> {
         nymd_client: Client<C>,
         caching_interval: Duration,
         cache: ValidatorCache,
+        storage: Option<ValidatorApiStorage>,
     ) -> Self {
         ValidatorCacheRefresher {
             nymd_client,
             cache,
             caching_interval,
+            storage,
         }
     }
 
-    fn annotate_bond_with_details(
+    async fn get_uptime(&self, identity: &IdentityKey, epoch: Interval) -> Option<Uptime> {
+        self.storage
+            .as_ref()?
+            .get_average_mixnode_uptime_in_the_last_24hrs(identity, epoch.end_unix_timestamp())
+            .await
+            .ok()
+    }
+
+    async fn annotate_bond_with_details(
+        &self,
         mixnodes: Vec<MixNodeBond>,
         interval_reward_params: EpochRewardParams,
+        current_epoch: Interval,
+        epochs_in_interval: u64,
+        current_operator_base_cost: u64,
+        rewarded_set_identities: &HashMap<IdentityKey, RewardedSetNodeStatus>,
     ) -> Vec<MixNodeBondAnnotated> {
-        mixnodes
-            .into_iter()
-            .map(|mixnode_bond| {
-                let stake_saturation = mixnode_bond
-                    .stake_saturation(
-                        interval_reward_params.staking_supply(),
-                        interval_reward_params.rewarded_set_size() as u32,
-                    )
-                    .to_num();
-                MixNodeBondAnnotated {
-                    mixnode_bond,
-                    stake_saturation,
-                }
-            })
-            .collect()
+        let mut annotated = Vec::new();
+        for mixnode_bond in mixnodes {
+            let stake_saturation = mixnode_bond
+                .stake_saturation(
+                    interval_reward_params.staking_supply(),
+                    interval_reward_params.rewarded_set_size() as u32,
+                )
+                .to_num();
+
+            let uptime = self
+                .get_uptime(mixnode_bond.identity(), current_epoch)
+                .await
+                .unwrap_or_default();
+
+            let is_active = rewarded_set_identities
```

Comment:
if mixnode is in the rewarded set it doesn't mean it's active; it can still be either active or in standby